### PR TITLE
address some bandit security recommendations

### PR DESCRIPTION
**Pull request by MartinRyan**
duplicate of - address some bandit security recommendations #2949

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Hi, looks like you've merged the Python3 master into this branch. Is this still targeted at 7.8.x?

**Comment by hjoliver** [comment]
@MartinRyan - minor conflicts here now.

**Comment by hjoliver** [comment]
@MartinRyan - I just saw @oliver-sanders' question above: 

> Hi, looks like you've merged the Python3 master into this branch. Is this still targeted at 7.8.x?

You didn't respond to this, and a quick look shows you do have a bunch of commits on this branch from master, which should not be here.  This is supposed to be your security changes on top of 7.8.x.

It's late here now ... maybe @oliver-sanders or @matthewrmshin could suggest a way out of this ... maybe an interactive rebase onto 7.8.x during which you delete all spurious commits (hopefully it is easy to identify them!) ??

**Comment by hjoliver** [comment]
@MartinRyan - I have pulled your branch myself and rebased it onto 7.8.x, removing all commits by others (which belong to master), and attempting to address conflicts during the rebase. I'll post a pointer to it shortly so you can check it before I do a force-push to overwrite this PR branch. I expect travis config might need some tweaking as a bunch of difficult conflicts occurred there).

BTW, for future reference, for a PR with a large number of commits on it like this, we prefer rebasing to merging from upstream (to update the your feature branch with new changes from the target branch) because it makes a clean linear history with no merge commits.

**Comment by hjoliver** [comment]
@MartinRyan - here's the diff with 7.8.x: 

https://github.com/cylc/cylc/compare/7.8.x...hjoliver:martin-c7?expand=1

Can you see if all the diffs look right and nothing important has been missed?  If it is good, I will force-push to your branch so we can use this same PR.

There's only one test failing on my rebased branch on T-CI, but not in my local environment.

**Comment by hjoliver** [comment]
And the test is a known flaky one.

**Comment by hjoliver** [comment]
OK, all tests pass on T-CI on my branch.  @MartinRyan - let me know when/if you agree to the force push.

**Comment by MartinRyan** [comment]
> @MartinRyan - I just saw @oliver-sanders' question above:
> 
> > Hi, looks like you've merged the Python3 master into this branch. Is this still targeted at 7.8.x?
> 
> You didn't respond to this, and a quick look shows you do have a bunch of commits on this branch from master, which should not be here. This is supposed to be your security changes on top of 7.8.x.


Ah yes, I  did see this and reverted this pull request branch to a previous commit using a force push from  6acb121 to e148921 and all tests passed at that time. I wasn't aware pf the rebasing preferences, I'll be sure to do that next time.

**Comment by MartinRyan** [comment]
Ok with the force push Hilary!

**Comment by hjoliver** [comment]
> Ah yes, I did see this and reverted this pull request branch to a previous commit using a force push from 6acb121 to e148921 and all tests passed at that time.

OK, interesting that the tests passed there, but if you look back in the list of commits the spurious changes from master were merged before #e148921, so there was still a bunch of "wrong code" on the branch (I guess the code from master brought its own corresponding test changes with it).

Anyhow, force push coming now...

**Comment by hjoliver** [comment]
Master commits expunged!

If Travis CI fails any tests, it ought to be just the flaky ones (this branch passed in my fork).

@matthewrmshin - can you please review this PR now?  I think we have previously discussed the changes to Jinja2 and Cherrypy and agree they are OK under the circumstances (which is, security improvements to bundled libraries that we're dropping from cylc-8, just to see out the reign of cylc-7).

**Comment by hjoliver** [comment]
(and this is otherwise the same as #2949)

File: .travis.yml [hunk 1]
```diff
@@ -84,4 +84,4 @@ jobs:
     script:
     - pycodestyle --ignore=E402,W503,W504 lib/cylc lib/Jinja2Filters/*.py lib/parsec/*.py $(grep -l '#!.*\<python\>' bin/*)
     - PYTHONPATH=$(pwd -P)/lib/ pytest --cov-append --cov=lib/cylc --cov=lib/parsec
-    after_script: true
+    after_script: true
```
Comment (matthewrmshin) [inline .travis.yml]: Reset this change?

Comment (hjoliver) [reply .travis.yml]: This one is probably my fault, during rebase conflict resolution. (It looks familiar).

**Comment by matthewrmshin** [comment]
In conflict again. Otherwise good to go.

**Pull request merged**